### ISSUE TITLE
Bug in beam search if the 'UNK' token is not in the vocabulary

### ISSUE THREAD

**Issue by deuscovrigus** [ruotianluo/self-critical.pytorch#123]
Hey ruotianluo!

I have found that the prerpo_labels.py script only adds the UNK token to the vocabulary if there are words with counts not exceeding the word threshold, according to:
(https://github.com/ruotianluo/self-critical.pytorch/blob/9b36d9bf1cad47cd8c917c895d926b3ff3d3e5e8/scripts/prepro_labels.py#L81)
However if you're using something other than coco like your own caption labels bad_count might be 0 , UNK is never the last label of the vocabulary. This causes a problem here in the beam search:
(https://github.com/ruotianluo/self-critical.pytorch/blob/9b36d9bf1cad47cd8c917c895d926b3ff3d3e5e8/models/CaptionModel.py#L143)
Basically if UNK is not the last token you will supress the last word in your vocabulary so when you're evaluating using beam_search that label will never be produced.

**Comment by ruotianluo** [comment]
You are absolutely correct.

**Comment by ruotianluo** [comment]
fixed in d83a04
